pkg/export/export_test.go

```diff
@@ -387,3 +387,64 @@ func TestExporter_drainBacklog(t *testing.T) {
 		t.Fatalf("got %d, want %d", got, want)
 	}
 }
+
+func TestExporter_shutdown(t *testing.T) {
+	var (
+		srv          = grpc.NewServer()
+		listener     = bufconn.Listen(1e6)
+		metricServer = &testMetricService{}
+	)
+	monitoring_pb.RegisterMetricServiceServer(srv, metricServer)
+
+	go func() { srv.Serve(listener) }()
+	defer srv.Stop()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	bufDialer := func(context.Context, string) (net.Conn, error) {
+		return listener.Dial()
+	}
+	metricClient, err := monitoring.NewMetricClient(ctx,
+		option.WithoutAuthentication(),
+		option.WithGRPCDialOption(grpc.WithInsecure()),
+		option.WithGRPCDialOption(grpc.WithContextDialer(bufDialer)),
+	)
+	if err != nil {
+		t.Fatalf("creating metric client failed: %s", err)
+	}
+
+	e, err := New(log.NewLogfmtLogger(log.NewSyncWriter(os.Stderr)), nil, ExporterOpts{})
+	if err != nil {
+		t.Fatalf("Creating Exporter failed: %s", err)
+	}
+
+	e.metricClient = metricClient
+	e.SetLabelsByIDFunc(func(i uint64) labels.Labels {
+		return labels.FromStrings("project_id", "test", "location", "test")
+	})
+
+	exportCtx, cancelExport := context.WithCancel(context.Background())
+
+	for i := 0; i < 50; i++ {
```

Comment:
Maxes out on 2048 samples. I'm assuming because they only go to one shard. Do you know how to create the shard in a different way?